cmd/dep/graphviz.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"hash/fnv"
+	"strings"
+	"bytes"
+	"fmt"
+)
+
+type graphviz struct {
+	ps []*gvnode
+	b  bytes.Buffer
+	h  map[string]uint32
+}
+
+type gvnode struct {
+	project  string
+	version  string
+	children []string
+}
+
+func (g graphviz) New() *graphviz {
+	ga := &graphviz{
+		ps: []*gvnode{},
+		h:  make(map[string]uint32),
+	}
+	return ga
+}
+
+func (g graphviz) output() bytes.Buffer {
+	g.b.WriteString("digraph { node [shape=box]; ")
+
+	for _, gvp := range g.ps {
+		g.h[gvp.project] = gvp.hash()
+
+		// Create node string
+		g.b.WriteString(fmt.Sprintf("%d [label=\"%s\"];", gvp.hash(), gvp.label()))
+	}
+
+	// Store relations to avoid duplication
+	rels := make(map[string]bool)
+
+	// Create relations
+	for _, dp := range g.ps {
+		for _, bsc := range dp.children {
+			for pr, hsh := range g.h {
+				if strings.HasPrefix(bsc, pr) {
```

Comment:
Let's keep it as a helper just in graphviz.go for now. If/when we see wider use for it, we can move it into a more general location.